src/CoreWCF.Primitives/src/CoreWCF/Channels/SecurityBindingElement.cs

```diff
@@ -0,0 +1,995 @@
+
+                if (this.LocalServiceSettings.DetectReplays)
+                {
+                    factory.NonceCache = this.LocalServiceSettings.NonceCache;
+                }
+            }
+
+            factory.SecurityBindingElement = (SecurityBindingElement)this.Clone();
+            factory.SecurityBindingElement.SetIssuerBindingContextIfRequired(issuerBindingContext);
+            factory.SecurityTokenManager = credentialsManager.CreateSecurityTokenManager();
+            SecurityTokenSerializer tokenSerializer = factory.SecurityTokenManager.CreateSecurityTokenSerializer(this.messageSecurityVersion.SecurityTokenVersion);
+            factory.StandardsManager = new SecurityStandardsManager(this.messageSecurityVersion, tokenSerializer);
+        }
+
+        internal abstract SecurityProtocolFactory CreateSecurityProtocolFactory<TChannel>(BindingContext context, SecurityCredentialsManager credentialsManager,
+        bool isForService, BindingContext issuanceBindingContext);
+
+        public override IServiceDispatcher BuildServiceDispatcher<TChannel>(BindingContext context, IServiceDispatcher innerDispatcher)
+        {
+            if (context == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("context");
+
+            if (!this.CanBuildServiceDispatcher<TChannel>(context))
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new ArgumentException(SR.Format(SR.ChannelTypeNotSupported, typeof(TChannel)), "TChannel"));
+            }
+
+            this.readerQuotas = context.GetInnerProperty<XmlDictionaryReaderQuotas>();
+            if (readerQuotas == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.EncodingBindingElementDoesNotHandleReaderQuotas)));
+            }
+
+            TransportBindingElement transportBindingElement = null;
+            if (context.RemainingBindingElements != null)
+                transportBindingElement = context.RemainingBindingElements.Find<TransportBindingElement>();
+
+            if (transportBindingElement != null)
+                this.maxReceivedMessageSize = transportBindingElement.MaxReceivedMessageSize;
+            return this.BuildServiceDispatcherCore<TChannel>(context, innerDispatcher);
+        }
+        protected abstract IServiceDispatcher BuildServiceDispatcherCore<TChannel>(BindingContext context, IServiceDispatcher serviceDispatcher)
+            where TChannel : class, IChannel;
+        public override bool CanBuildServiceDispatcher<TChannel>(BindingContext context)
+        {
+            if (context == null)
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("context");
+
+            // InternalDuplexBindingElement.AddDuplexListenerSupport(context, ref this.internalDuplexBindingElement);
+
+            //  if (this.SessionMode)
+            //  {
```

Comment:
This is required for SecureConversation support, how did you avoid implementing this?

Reply:
going to debug and resolve...